### PR TITLE
Update analytics examples to not use provided url

### PR DESCRIPTION
**Pull request by Janpot**
Use actual value, independent from `basePath`

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| buildDuration | 11.4s | 11.6s | ⚠️ +237ms |
| nodeModulesSize | 66.5 MB | 66.5 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.922 | 1.882 | -0.04 |
| / avg req/sec | 1300.53 | 1328.38 | +27.85 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.196 | 1.146 | -0.05 |
| /error-in-render avg req/sec | 2090.94 | 2180.88 | +89.94 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| 677f882d2ed8..db49.js gzip | 9.7 kB | 9.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-37681a1..8579.js gzip | 6.72 kB | 6.72 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.1 kB | 87.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.1 kB | 6.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-a93d67d..dule.js gzip | 5.79 kB | 5.79 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8e69b..8695.js gzip | 3.41 kB | 3.41 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.66 kB | 7.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-663e6..dule.js gzip | 2.24 kB | 2.24 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.29 kB | 5.29 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| index.html gzip | 945 B | 945 B | ✓ |
| link.html gzip | 954 B | 954 B | ✓ |
| withRouter.html gzip | 939 B | 939 B | ✓ |
| Overall change | 2.84 kB | 2.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| buildDuration | 12.3s | 12.3s | -29ms |
| nodeModulesSize | 66.5 MB | 66.5 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| 677f882d2ed8..db49.js gzip | 9.7 kB | 9.7 kB | ✓ |
| framework.HASH.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-37681a1..8579.js gzip | 6.72 kB | 6.72 kB | ✓ |
| polyfills-05..1236.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-488d..c0e7.js gzip | 751 B | 751 B | ✓ |
| Overall change | 87.1 kB | 87.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 6.1 kB | 6.1 kB | ✓ |
| framework.HA..dule.js gzip | 39.1 kB | 39.1 kB | ✓ |
| main-a93d67d..dule.js gzip | 5.79 kB | 5.79 kB | ✓ |
| polyfills-05..dule.js gzip | 30.8 kB | 30.8 kB | ✓ |
| webpack-4f62..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 82.6 kB | 82.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _app-8f5f611..1f7b.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-8e69b..8695.js gzip | 3.41 kB | 3.41 kB | ✓ |
| hooks-f7f3d0..7465.js gzip | 887 B | 887 B | ✓ |
| index-08fb3f..c0e9.js gzip | 227 B | 227 B | ✓ |
| link-ddd176e..5566.js gzip | 1.29 kB | 1.29 kB | ✓ |
| routerDirect..8aa1.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..e777.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.66 kB | 7.66 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _app-669dbe5..dule.js gzip | 626 B | 626 B | ✓ |
| _error-663e6..dule.js gzip | 2.24 kB | 2.24 kB | ✓ |
| hooks-805c40..dule.js gzip | 387 B | 387 B | ✓ |
| index-6ba5a4..dule.js gzip | 226 B | 226 B | ✓ |
| link-69bc264..dule.js gzip | 1.25 kB | 1.25 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-d..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.29 kB | 5.29 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _buildManifest.js gzip | 275 B | 275 B | ✓ |
| _buildManife..dule.js gzip | 281 B | 281 B | ✓ |
| Overall change | 556 B | 556 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Janpot/next.js router-events-example | Change |
| - | - | - | - |
| _error.js | 881 kB | 881 kB | ✓ |
| 404.html | 4.18 kB | 4.18 kB | ✓ |
| hooks.html | 3.82 kB | 3.82 kB | ✓ |
| index.js | 882 kB | 882 kB | ✓ |
| link.js | 923 kB | 923 kB | ✓ |
| routerDirect.js | 917 kB | 917 kB | ✓ |
| withRouter.js | 917 kB | 917 kB | ✓ |
| Overall change | 4.53 MB | 4.53 MB | ✓ |

</details>

</details>
Commit: 1b9a6843722aabd81ac4959778c512473224462c

**Comment by paambaati** [comment]
@Janpot For just a user (not a contributor yet), would you mind explaining why using `basePath` isn’t preferred here? I ask for my own edification.

**Comment by Janpot** [comment]
@paambaati `basePath` _is_ preferred. This PR makes sure that the correct path is always sent, regardless of whether router events pass urls with or without a `basePath`. The whole library works with urls relative to `basePath`. There's the idea to make router events urls root-relative, because these are often used for analytics. The PR here instead proposes to update the examples so that we consistency can be maintained between all urls in next.js.

File: examples/with-google-analytics-amp/lib/gtag.js [hunk 1]
```diff
@@ -1,9 +1,9 @@
 export const GA_TRACKING_ID = '<YOUR_GA_TRACKING_ID>'
 
 // https://developers.google.com/analytics/devguides/collection/gtagjs/pages
-export const pageview = (url) => {
+export const pageview = () => {
   window.gtag('config', GA_TRACKING_ID, {
-    page_path: url,
+    page_path: window.location.pathname + window.location.search,
```
Comment (timneutkens) [inline examples/with-google-analytics-amp/lib/gtag.js:6]: Is this missing the window.location.hash 🤔

Comment (Janpot) [reply examples/with-google-analytics-amp/lib/gtag.js:6]: Good question, I took inspiration from [`react-ga`](https://github.com/react-ga/react-ga/blob/e82caf57f0c797bc801db6057cfc3ba43a1a3a2d/README.md#with-npm) for this.

**Pull request closed without merging**